autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgentTest.java

```diff
@@ -0,0 +1,665 @@
+        Mockito.doReturn(new HostVO("")).when(hostService).findHostById(Mockito.anyLong());
+
+        administrationAgent.migrateVM(vmInstanceVo, hostVo);
+
+        Mockito.verify(userVmService, Mockito.times(0)).migrateVirtualMachine(Mockito.anyLong(), Mockito.any(HostVO.class));
+    }
+
+    @Test
+    public void migrateVMTestTargetHostDifferentFromVmHost() throws Exception {
+        VMInstanceVO vmInstanceVo = migrateVMSetupVmInstanceVo(0l);
+        HostVO hostVo = migrateVMSetupHostVo(1l);
+        Mockito.doReturn(vmInstanceVo).when(userVmService).migrateVirtualMachine(Mockito.anyLong(), Mockito.any(HostVO.class));
+
+        administrationAgent.migrateVM(vmInstanceVo, hostVo);
+
+        Mockito.verify(userVmService, Mockito.times(1)).migrateVirtualMachine(Mockito.anyLong(), Mockito.any(HostVO.class));
+    }
+
+    @Test
+    public void shutdownHostTestHostWithoutVms() {
+        setupShutdownHostTest(new ArrayList<VMInstanceVO>());
+        administrationAgent.shutdownHost(0l);
+        verifyShutdownHostCalledMethods(1);
+    }
+
+    @Test
+    public void shutdownHostTestHostWithVms() {
+        List<VMInstanceVO> vms = new ArrayList<>();
+        VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
+        vms.add(vm);
+        setupShutdownHostTest(vms);
+
+        administrationAgent.shutdownHost(0l);
+
+        verifyShutdownHostCalledMethods(0);
+    }
+
+    @Test
+    public void getClusterUpHostsTest() {
+        ClusterVO cluster = createCluster(null);
+        List<HostVO> hosts = setupGetClusterUpHostsTest(2);
+
+        List<HostResources> result = administrationAgent.getClusterUpHosts(cluster);
+
+        Assert.assertEquals(hosts.size(), result.size());
+        verifyGetClusterUpHostsCalledMethods(2);
+    }
+
+    @Test
+    public void getClusterUpHostsTestNoHostRunning() {
+        ClusterVO cluster = createCluster(null);
+        List<HostVO> hosts = setupGetClusterUpHostsTest(0);
+
+        List<HostResources> result = administrationAgent.getClusterUpHosts(cluster);
+
+        Assert.assertEquals(hosts.size(), result.size());
+        verifyGetClusterUpHostsCalledMethods(0);
+    }
+
+    @Test
+    public void getClusterIdleHostsTest() {
+        ClusterVO cluster = createCluster(null);
+        List<HostResources> hosts = createHosts(2);
+        List<VmResources> vmsResources = new ArrayList<>();
+        vmsResources.add(new VmResources(0l, 1, 1000, 1000));
+        vmsResources.add(new VmResources(0l, 1, 1000, 1000));
+        hosts.get(0).setVmsResources(vmsResources);
+
+        Mockito.doReturn(hosts).when(administrationAgent).getClusterUpHosts(Mockito.any(ClusterVO.class));
+
+        List<HostResources> result = administrationAgent.getClusterIdleHosts(cluster);
+
+        Mockito.verify(administrationAgent).getClusterUpHosts(Mockito.any(ClusterVO.class));
+        Assert.assertEquals(1, result.size());
```

Comment:
Here you should also check if the host that was returned is in fact the idle one.